az-core/src/main/java/azkaban/metrics/MetricsManager.java

```diff
@@ -114,27 +115,27 @@ public Timer addTimer(final String name) {
    * reporting metrics to remote metrics collector. Note: this method must be synchronized, since
    * both web server and executor will call it during initialization.
    */
-  public synchronized void startReporting(final String reporterName, final Props props) {
+  public synchronized void startReporting(final Props props) {
     final String metricsReporterClassName = props.get(CUSTOM_METRICS_REPORTER_CLASS_NAME);
-    final String metricsServerURL = props.get(METRICS_SERVER_URL);
-    if (metricsReporterClassName != null && metricsServerURL != null) {
+    final String metricsReporterConfigPath =
+        props.get(CUSTOM_METRICS_REPORTER_CONFIG_PATH); // optional param
+    if (!StringUtils.isBlank(metricsReporterClassName)) {
       try {
-        log.info("metricsReporterClassName: " + metricsReporterClassName);
+        log.info("The class name of the custom metrics reporter is: {}", metricsReporterClassName);
         final Class<?> metricsClass = Class.forName(metricsReporterClassName);
 
-        final Constructor<?> ctor = metricsClass.getConstructor(reporterName.getClass(),
-            this.registry.getClass(), metricsServerURL.getClass(), boolean.class);
-        ctor.newInstance(reporterName, this.registry, metricsServerURL, true);
+        final Constructor<?> ctor = metricsClass.getConstructor(this.registry.getClass(),
+            metricsReporterConfigPath.getClass());
+        ctor.newInstance(this.registry, metricsReporterConfigPath);
 
       } catch (final Exception e) {
-        log.error("Encountered error while loading and instantiating "
-            + metricsReporterClassName, e);
-        throw new IllegalStateException("Encountered error while loading and instantiating "
-            + metricsReporterClassName, e);
+        final String errMsg = "Encountered an error while loading or instantiating the custom "
```

Comment:
If possible, try to keep complete text message in one line. It will be easy to search.

Reply:
you're right, done.